src/module_compilation.ml

```diff
@@ -78,7 +78,7 @@ let build_cm sctx ?sandbox ~dynlink ~flags ~cm_kind ~(dep_graph:Ocamldep.dep_gra
          Build.dyn_paths (Build.arr (lib_dependencies ~cm_kind)) >>>
          Build.run ~context:ctx (Dep compiler)
            ~extra_targets
-           [ Ocaml_flags.get_for_cm flags ~cm_kind
+           [ Ocaml_flags.get_for_cm flags ~target:m.impl.name ~cm_kind
```

Comment:
So, this is matching on the filename, rather than the module name as what `(preprocess (per_file ...))` currently does. This seems fine, since essentially that's what `per_file` suggests.

However, this is not going to work right now since `Per_file` builds a map from module names, i.e. without the extension and capitalized. Moreover, at this point, if you use preprocessing, the filename won't even be the `.ml` or `.mli`. It will be `.pp.ml` or `.pp.mli`.